From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Shelley Vohr <shelley.vohr@gmail.com>
Date: Wed, 22 Oct 2025 11:03:57 +0200
Subject: feat: disable js source phase imports by default.

Refs:
- https://chromium-review.googlesource.com/c/v8/v8/+/7003082
- https://chromium-review.googlesource.com/c/chromium/src/+/6336964
- https://github.com/nodejs/node/pull/56919

We need to disable source phase imports in renderer and worker processes - Chromium
disables them in content/renderer/render_process_impl.cc via and the process will
hard crash otherwise.

They shouldn't be enabled by default in Node.js either as they are still not yet
Stage 3.

Upstreamed in https://github.com/nodejs/node/pull/60364

diff --git a/src/node.cc b/src/node.cc
index 5713d49d859e1161e1d6703c0b6f3d717a5a9a34..829634c084cb91eb7f488dbdd48175a093dd6e12 100644
--- a/src/node.cc
+++ b/src/node.cc
@@ -780,7 +780,7 @@ static ExitCode ProcessGlobalArgsInternal(std::vector<std::string>* args,
     env_opts->abort_on_uncaught_exception = true;
   }
 
-  v8_args.emplace_back("--js-source-phase-imports");
+  // v8_args.emplace_back("--js-source-phase-imports");
 
 #ifdef __POSIX__
   // Block SIGPROF signals when sleeping in epoll_wait/kevent/etc.  Avoids the
